### PR TITLE
PERSIST_INSTALL flag for e2e tests

### PR DESCRIPTION
**Pull request by sheidkamp**
# Description
* Make the old and new style e2e tests use consistent flags/env variables to control Gloo install and uninstall
  * SKIP_INSTALL - Skips the Gloo install if `true`. Defaults to `false`
  * TEAR_DOWN - Uninstalls Gloo after the test run if `true`. Defaults to true for new style e2e tests, false for the old style tests
  * PERSIST_INSTALL - If `true` uses the existing Gloo install if present, and installs Gloo otherwise, and leaves Gloo installed after the test tun. Defaults to `false`

The new `PERSIST_INSTALL` flag was created so a single command can be used to run tests repeatedly without having to  run a separate setup step

## Code changes
Test code and utility changes to support flags

## Docs changes
Developer facing docs updated to reflect changes

## Interesting decisions
We chose to leave the existing flags in place as they define specific behavior that may still be useful.

### Kubernetes e2e tests
Assumes test cluster is created with no Gloo installed

Initial run with PERSIST_INSTALL flag, should install and leave installation:
```
go clean -testcache; PERSIST_INSTALL=true GOLANG_PROTOBUF_REGISTRATION_CONFLICT=ignore go test -failfast -count 1 -v -timeout 450s ./test/kubernetes/e2e/tests -run ^TestHelm$/^Helm$/^TestApplyCRDs$
```
Re-run same command, should use the existing install and leave installation

Get the same behavior with the SKIP_INSTALL and TEAR_DOWN flags:
```
go clean -testcache; SKIP_INSTALL=true TEAR_DOWN=false GOLANG_PROTOBUF_REGISTRATION_CONFLICT=ignore go test -failfast -count 1 -v -timeout 450s ./test/kubernetes/e2e/tests -run ^TestHelm$/^Helm$/^TestApplyCRDs$
```

Test tear down:
```
go clean -testcache; SKIP_INSTALL=true TEAR_DOWN=true GOLANG_PROTOBUF_REGISTRATION_CONFLICT=ignore go test -failfast -count 1 -v -timeout 450s ./test/kubernetes/e2e/tests -run ^TestHelm$/^Helm$/^TestApplyCRDs$
```

At this point Gloo should be uninstalled

### Kube2e tests
This will run faster if you focus a Gateway test such as [It("works", ...)](https://github.com/solo-io/gloo/blob/main/test/kube2e/gateway/gateway_test.go#L1595)

Initial run with PERSIST_INSTALL flag, should install and leave installation:
```
PERSIST_INSTALL=true KUBE2E_TESTS=gateway make run-kube-e2e-test
```

Re-run same command, should use the existing install and leave installation

Get the same behavior with the SKIP_INSTALL and TEAR_DOWN flags:
```
SKIP_INSTALL=true TEAR_DOWN=false KUBE2E_TESTS=gateway make run-kube-e2e-test
```

Test tear down:
```
SKIP_INSTALL=true TEAR_DOWN=true KUBE2E_TESTS=gateway make run-kube-e2e-test
```

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

BOT NOTES: 
resolves https://github.com/solo-io/solo-projects/issues/7432

### CODE REVIEW

File: pkg/utils/envutils/env.go [hunk 1]
```diff
@@ -11,6 +11,11 @@ func IsEnvTruthy(envVarName string) bool {
 	return IsTruthyValue(os.Getenv(envVarName))
 }
 
+// EnvVal returns the value of an environment variable
+func GetEnv(envVarName string) string {
+	return os.Getenv(envVarName)
+}
+
```
Comment (ryanrolds) [inline pkg/utils/envutils/env.go:18]: nit: I'm new and if this is a pattern we use at Solo, please ignore this comment. 

This has the same signature and doesn't change the result, it looks unnecessary. If I was working on something and saw an `GetEnv` call I would have to go look at this function to what it's doing only to find it isn't doing anything. Calling `os.Getenv` instead wouldn't have that overhead.

File: test/testutils/env.go [hunk 2]
```diff
@@ -127,3 +157,8 @@ func IsEnvTruthy(envVarName string) bool {
 func IsEnvDefined(envVarName string) bool {
 	return envutils.IsEnvDefined(envVarName)
 }
+
+// GetEnv returns the value of an environment variable
+func GetEnv(key string) string {
+	return envutils.GetEnv(key)
+}
```
Comment (ryanrolds) [inline test/testutils/env.go:164]: Same nit as above.

File: test/kubernetes/testutils/assertions/glooctl.go [hunk 3]
```diff
@@ -13,22 +13,27 @@ import (
...
+		Metadata: core.Metadata{
+			Namespace: p.glooGatewayContext.InstallNamespace,
+		},
+		Top: options.Top{
+			Ctx: contextWithCancel,
+		},
+	}
+	return check.CheckResources(contextWithCancel, printers.P{}, opts)
+}
+
```
Comment (ryanrolds) [inline test/kubernetes/testutils/assertions/glooctl.go:30]: Should we DRY this and the other copy out?

**Review by andy-fong — APPROVED** [review]

**Review by ryanrolds — APPROVED** [review]
Looks good. My comments are not blockers.

File: pkg/utils/envutils/env.go (hunk 1 above)
Comment (sam-heilbron) [reply pkg/utils/envutils/env.go:18]: I agree with this

**Pull request merged**